lib/app/preflight_check.rb

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'app/version'
+require 'utility/logger'
+
+module App
+  class PreflightCheck
+    class CheckFailure < StandardError; end
+
+    CONNECTORS_INDEX = '.elastic-connectors'
+    JOB_INDEX = '.elastic-connectors-sync-jobs'
+
+    def self.run!
+      App::PreflightCheck.new.run!
+    end
+
+    def run!
+      check_es_connection!
+      check_es_version!
+      check_system_indices!
+    end
+
+    private
+
+    #-------------------------------------------------------------------------------------------------
+    # Checks to make sure we can connect to Elasticsearch and make API requests to it
+    def check_es_connection!
+      response = client.cluster.health
+      case response['status']
+      when 'green'
+        Utility::Logger.info('Elasticsearch backend is running and healthy.')
+      when 'yellow'
+        Utility::Logger.warn('Elasticsearch backend is running but the status is yellow.')
+      when 'red'
+        fail_check!('Elasticsearch backend is running but is unhealthy.')
+      else
+        fail_check!("Unexpected cluster status: #{response['status']}")
+      end
+    rescue StandardError
+      fail_check!('Could not connect to Elasticsearch backend. Make sure it is running and healthy.')
+    end
+
+    #-------------------------------------------------------------------------------------------------
+    # Ensures that the version of Elasticsearch is compatible with connector service
+    def check_es_version!
+      info = client.info
+      version = info.dig('version', 'number')
+      fail_check!("Cannot retrieve version from Elasticsearch response:\n#{info.to_json}") unless version
+
+      if match_es_version?(version)
```

Comment:
I only checked if `Major.Minor` matches. There is more comprehensive [version check in ent-search](https://github.com/elastic/ent-search/blob/38279fff22c3da29b715a59a88684f1a13508a8d/shared_togo/lib/shared_togo/elasticsearch_checks.rb#L51-L61), which I'm not sure whether we want to adopt. @elastic/ingestion-team